Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component } from '@angular/core';
+import { isOverlapRecurrentAppointment } from '../utils/isOverlapRecurrentAppointment';
+import { appointments } from '../data/appointments';
+import { CloseButtonOptions, Appointment } from './interfaces';
+import { AppointmentAddingEvent } from 'devextreme/ui/scheduler';
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css'],
+})
+
+export class AppComponent {
+  dataSource: Appointment[] = appointments;
+  currentDate: Date = new Date(2022, 9, 1);
+  popupVisible: boolean = false;
+  closeButtonOptions: CloseButtonOptions  = {
+    text: 'Close',
+    onClick: () => {
+      this.popupVisible = false;
+    }
+  };
+
+  handleAppointmentActions(event: AppointmentAddingEvent, newAppointment: Appointment): void {
+    const recurrentAppointments = this.getRecurrentAppointments();
+    for(const recurrentAppointment of recurrentAppointments) {
+      const isOverlap = isOverlapRecurrentAppointment(
+        event,
+        recurrentAppointment,
+        newAppointment);
+      if (isOverlap) {
+        this.cancelAppointmentAdding(event);
+      }
+    }
+  }
```

Comment:
Let's split this method into two public methods to handle updating existing recurrent appointment case:

```suggestion
handleAppointmentAdd(
    event: AppointmentAddingEvent
  ): void {
    this.handleAppointmentActions(
      event,
      this.getRecurrentAppointments(),
      event.appointmentData as Appointment,
    );
  }

  handleAppointmentUpdate(
    event: AppointmentUpdatingEvent,
  ): void {
    const recurrentAppointments = this.getRecurrentAppointments()
      .filter((appointment) => appointment !== event.oldData);
    this.handleAppointmentActions(
      event,
      recurrentAppointments,
      event.newData as Appointment,
    )
  }

  private handleAppointmentActions(
    event: AppointmentAddingEvent | AppointmentUpdatingEvent,
    recurrentAppointments: Appointment[],
    newAppointment: Appointment): void {
    for(const recurrentAppointment of recurrentAppointments) {
      const isOverlap = isOverlapRecurrentAppointment(
        event,
        recurrentAppointment,
        newAppointment);
      if (isOverlap) {
        this.cancelAppointmentAdding(event);
      }
    }
  }
```